cram_urdf_projection/cram_urdf_projection/src/low-level.lisp

```diff
@@ -372,9 +388,9 @@
     (set-configuration :right right-configuration)))
 
 ;;; cartesian movement
-(defparameter *gripper-length* 0.2 "PR2's gripper length in meters, for calculating TCP -> EE")
+
 (defparameter *tcp-pose-in-ee* (cl-transforms:make-pose
-                                (cl-transforms:make-3d-vector (- *gripper-length*) 0 0)
+                                (cl-transforms:make-3d-vector (- (get-gripper-length)) 0 0)
```

Comment:
This will not work if the TCP is not on the X axis of EE, but on Y or Z axis. Please use a full vector.

Reply:
This is definitely not going to work, `defparameter`` gets evaluated at compile time, when the TF listener is not initialized yet. Did you test this? Solving the problem without using TF whatsoever will solve this issue with ``defparameter`` as well.

Reply:
deleted this for now since we searching for a better solution